### PR TITLE
CI fix 

### PR DESCRIPTION
**Pull request by krissetto**
Adds `fetch-depth:0` and `fetch-tags: true` to successfully get the repo tags, needed to properly build the library.

See https://github.com/docker/docker-py/pull/3259 attempts for more details

### CODE REVIEW

**Comment by krissetto** [comment]
Also updates the `setup-python` gh action to get rid of some node related warnings

**Comment by krissetto** [comment]
All failures are related to the requests 2.32+ changes, and will be fixed with https://github.com/docker/docker-py/pull/3257 after this CI fix gets merged

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -62,6 +62,9 @@ jobs:
 
         steps:
             - uses: actions/checkout@v4
+              with:
+                fetch-depth: 0
+                fetch-tags: true
             - name: make ${{ matrix.variant }}
```
Comment (alexanderankin) [inline .github/workflows/ci.yml:68]: not sure what the expected behavior here is:

```console
$ git clone --depth=0 git@github.com:docker/docker-py.git
fatal: depth 0 is not a positive number
```

Comment (krissetto) [reply .github/workflows/ci.yml:68]: it's a value that's valid for the gh action, it just means fetch all essentially

Comment (krissetto) [reply .github/workflows/ci.yml:68]: https://github.com/actions/checkout for reference

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
We verified all failures are unrelated to this change in https://github.com/docker/docker-py/pull/3259, which also contains these changes; bringing this one in so that we can rebase the other PR(s)

**Pull request merged**